### PR TITLE
Feature/sg 128 kd recipe resnet50

### PR DESCRIPTION
**Pull request by shanibenbaruch**
1. Add train from recipe for KD model.
2. Add a new KD Trainer.
3. update pre-trained models.

### CODE REVIEW

File: src/super_gradients/examples/train_from_recipe_example/train_from_kd_recipe.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""
```
Comment (shaydeci) [inline src/super_gradients/examples/train_from_recipe_example/train_from_kd_recipe.py:1]: can we move this file to a new folder (module) inside examples called "train_from_kd_recipe_example"?

File: src/super_gradients/__init__.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 from super_gradients.training import ARCHITECTURES, losses, utils, datasets_utils, DataAugmentation, \
-    TestDatasetInterface, SegmentationTestDatasetInterface, DetectionTestDatasetInterface, ClassificationTestDatasetInterface, SgModel
+    TestDatasetInterface, SegmentationTestDatasetInterface, DetectionTestDatasetInterface, ClassificationTestDatasetInterface, SgModel, KDModel
 from super_gradients.common import init_trainer, is_distributed
 from super_gradients.examples.train_from_recipe_example import train_from_recipe
```
Comment (shaydeci) [inline src/super_gradients/__init__.py:4]: Please also add an import for train_from_kd_recipe_example so users could run it from the command line without all the python path overhead.

**Review by shaydeci — CHANGES_REQUESTED** [review]
Minor comments inline, nice work 👍 .

File: src/super_gradients/__init__.py (hunk 2 above)
Comment (shanibenbaruch) [reply src/super_gradients/__init__.py:4]: Added in 5688491a77aff88cbc04113e6b79d54a8c88d8d2

File: src/super_gradients/recipes/imagenet_resnet50_kd.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
+#  ResNet50 Imagenet classification training:
+#  This example trains with batch_size = 192 * 8 GPUs, total 1536.
+#  Training time on 8 x GeForce RTX A5000 is 9min / epoch.
```
[the remark this answers is not in the document]
Comment (shanibenbaruch) [reply src/super_gradients/recipes/imagenet_resnet50_kd.yaml:3]: Done. thanks.

File: src/super_gradients/examples/train_from_recipe_example/train_from_kd_recipe.py (hunk 1 above)
Comment (shanibenbaruch) [reply src/super_gradients/examples/train_from_recipe_example/train_from_kd_recipe.py]: Done in 5688491a77aff88cbc04113e6b79d54a8c88d8d2

File: src/super_gradients/recipes/training_hyperparams/imagenet_resnet50_kd_train_params.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+defaults:
+  - default_train_params
+
+max_epochs: 310
```
[the remark this answers is not in the document]
Comment (shanibenbaruch) [reply src/super_gradients/recipes/training_hyperparams/imagenet_resnet50_kd_train_params.yaml]: updated

**Pull request merged**